plugins/nominal-connection-checker/src/type_hierarchy.js

```diff
@@ -326,6 +434,22 @@ class TypeDef {
     this.descendants_.add(descendantName);
   }
 
+  /**
+   * Adds the given parameter info to the list of parameters of this type.
+   * @param {string} paramName The caseless name of the parameter.
+   * @param {!Variance} variance The variance of the parameter.
+   * @param {number=} index The index to insert the parameter at. If undefined,
+   *     the parameter will be added at the end.
+   */
+  addParam(paramName, variance, index = undefined) {
+    const param = new ParamDef(paramName, variance);
+    if (index != undefined) {
+      this.params_.splice(index, param);
```

Comment:
This is missing the first optional parameter, `deleteCount`. Should be `this.params_.splice(index, 0, param)`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice#syntax

Reply:
Thanks for the catch!